libraries/chain/block_log.cpp

```diff
@@ -746,6 +748,25 @@ namespace eosio { namespace chain {
       my->head.reset();
    }
 
+   std::shared_ptr<std::vector<char>> detail::block_log_impl::read_block_stream_by_num(uint32_t block_num) {
+       shared_ptr<std::vector<char>> send_buff;
+
+
+       auto blocks_dir      = fc::canonical(block_file.get_file_path().parent_path());
+       const auto block_log_path  = blocks_dir / "blocks.log";
+       fc::cfile block_file;
+       block_file.set_file_path(block_log_path);
+       block_file.open(fc::cfile::update_rw_mode);
```

Comment:
If the plan is for this method to be called from any thread then, accessing `block_file` here is not thread safe.
Also I would add a comment to the header that the method is thread safe.